File: src/entity/Booking.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from "typeorm";
import { Property } from "./Property";
import { Users } from "./Users";

@Entity()
export class Booking {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    price: number;

    @Column("date", { name: "booking_date" })
    bookingDate;

    @Column("datetime", { name: "check_in" })
    checkIn;

    @Column("datetime", { name: "check_out" })
    checkOut;

    @ManyToOne(type => Users)
    @JoinColumn({ name: 'user_id' })
    users: Users

    @ManyToOne(type => Property)
    @JoinColumn({ name: 'property_id' })
    property: Property

    @CreateDateColumn({ name: "created_at" })
    createdAt;

    @UpdateDateColumn({ name: "updated_at" })
    updatedAt;
} 

File: demo/src/main/java/com/example/demo/entities/Booking.java
package com.example.demo.entities;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "booking")
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column()
    private Integer price;

    @Column(name="booking_date")
    private Date bookingDate;

    @Column(name="check_in")
    private Date checkIn;

    @Column(name="check_out" )
    private Date checkOut;

    @Column(name="created_at")
    private Date createdAt;

    @Column(name="updated_at")
    private Date updatedAt;


    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getPrice() {
        return this.price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public Date getBookingDate() {
        return this.bookingDate;
    }

    public void setBookingDate(Date bookingDate) {
        this.bookingDate = bookingDate;
    }

    public Date getCheckIn() {
        return this.checkIn;
    }

    public void setCheckIn(Date checkIn) {
        this.checkIn = checkIn;
    }

    public Date getCheckOut() {
        return this.checkOut;
    }

    public void setCheckOut(Date checkOut) {
        this.checkOut = checkOut;
    }

  

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return this.updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

}